addons/mail/models/mail_link_preview.py

```diff
@@ -68,15 +62,22 @@ def _is_link_preview_enabled(self):
         return link_preview_throttle > 0
 
     @api.model
-    def _is_domain_throttled(self, url):
-        domain = urlparse(url).netloc
-        date_interval = fields.Datetime.to_string((datetime.now() - relativedelta(seconds=10)))
-        call_counter = self.search_count([
-            ('source_url', 'ilike', domain),
-            ('create_date', '>', date_interval),
-        ])
-        link_preview_throttle = int(self.env['ir.config_parameter'].get_param('mail.link_preview_throttle', 99))
-        return call_counter > link_preview_throttle
+    def _get_link_preview_from_url(self, url):
```

Comment:
Maybe we could align naming on other "find if exists else create" methods, like (and we don't have to repeat 'link preview' namespace as it is on model)
```suggestion
    def _search_or_create_from_url(self, url):
```